models/record.go

```diff
@@ -95,40 +95,65 @@ type RecordConfig struct {
 
 	// If you add a field to this struct, also add it to the list in the UnmarshalJSON function.
 	MxPreference     uint16            `json:"mxpreference,omitempty"`
+
 	SrvPriority      uint16            `json:"srvpriority,omitempty"`
 	SrvWeight        uint16            `json:"srvweight,omitempty"`
 	SrvPort          uint16            `json:"srvport,omitempty"`
+
 	CaaTag           string            `json:"caatag,omitempty"`
 	CaaFlag          uint8             `json:"caaflag,omitempty"`
+
 	DsKeyTag         uint16            `json:"dskeytag,omitempty"`
 	DsAlgorithm      uint8             `json:"dsalgorithm,omitempty"`
 	DsDigestType     uint8             `json:"dsdigesttype,omitempty"`
 	DsDigest         string            `json:"dsdigest,omitempty"`
+
 	LocVersion       uint8             `json:"locversion,omitempty"`
 	LocSize          uint8             `json:"locsize,omitempty"`
 	LocHorizPre      uint8             `json:"lochorizpre,omitempty"`
 	LocVertPre       uint8             `json:"locvertpre,omitempty"`
 	LocLatitude      uint32            `json:"loclatitude,omitempty"`
 	LocLongitude     uint32            `json:"loclongitude,omitempty"`
 	LocAltitude      uint32            `json:"localtitude,omitempty"`
+
+	LocLatDegrees    uint8             `json:"loclatdegrees,omitempty"`
+	LocLatMinutes    uint8             `json:"lotlatminutes,omitempty"`
+	LocLatSeconds    float32           `json:"loclatseconds,omitempty"`
+	LocLatDirection  string            `json:"loclatdirection,omitempty"`
+	LocLongDegrees   uint8             `json:"loclongdegrees,omitempty"`
+	LocLongMinutes   uint8             `json:"lotlongminutes,omitempty"`
+	LocLongSeconds   float32           `json:"loclongseconds,omitempty"`
+	LocLongDirection string            `json:"loclongdirection,omitempty"`
+
+	LocOrigAltitude  int32             `json:"locorigaltitude,omitempty"`
+	LocOrigSize      float32           `json:"locorigsize,omitempty"`
+	LocOrigHorizPre  float32           `json:"locorighorizpre,omitempty"`
+	LocOrigVertPre   float32           `json:"locorigvertpre,omitempty"`
```

Comment:
Why does this provider need new fields when multiple providers already support LOC without needing the extra fields?

Reply:
The Cloudflare API requires the original degrees, minutes, seconds values whereas to my knowledge the LOC model converts to decimal degrees, which doesn't fit into the request data Cloudflare asks for.
I'm trying to pass through both the DD and the original DMS values to the provider to allow either to be used.

https://developers.cloudflare.com/api/operations/dns-records-for-a-zone-create-dns-record (Body -> LOC Record)